46 create events page

catbus00 opened the pull request:
### Attached Issue

This PR resolves #46 

## Change Summary
Add component AddEventToGroupForm
Create Event form for valid user to create when logged in with a group

## I Had Difficulty With

credentials and auth

Commit:
added account creation and authenticated content

dargaCode committed:
Merge branch 'dev' into #27_Account_creation_page

dargaCode committed:
Merge remote-tracking branch 'origin/dev' into #27_Account_creation_page

Commit:
connected to the API from the backend

Commit:
Merge branch '#23_Account_creation_page' of https://github.com/Code-the-Dream-School/ee-prac-team2-front into #27_Account_creation_page

Victoria240 committed:
Prettier and ES Lint test

Victoria240 committed:
Update package-lock.json

Victoria240 committed:
Update package-lock.json

Victoria240 committed:
Update package.json

Victoria240 committed:
Changed the file structure, made fields required andadded min length for password

Victoria240 committed:
lifted the state by maintaining it in the AccountCreationContainer file instead of AccountCreation

Victoria240 committed:
created group components

Victoria240 committed:
Merge branch '#27_Account_creation_page' of https://github.com/Code-the-Dream-School/ee-prac-team2-front into #42_Create-Group-Page

Victoria240 committed:
Fixed merge conflict

Victoria240 committed:
minor fixes to AccountCreationContainer

Victoria240 committed:
Merge branch '#27_Account_creation_page' of https://github.com/Code-the-Dream-School/ee-prac-team2-front into #42_Create-Group-Page

Victoria240 committed:
made a request to users api

catbus00 committed:
Sync with Annie's changes

Victoria240 committed:
add checkboxes to list of users

catbus00 committed:
events form without POST request

angelkbrown committed:
Fix typo and use env var for api url

catbus00 committed:
merge branch to include auth container and update activity api schema changes

catbus00 committed:
merge branch to include auth container and update activity api schema changes

catbus00 committed:
convert date time to iso8601

angelkbrown committed:
Use environment variable for API endpoint, use fetch instead of axios, add headers for credentials

catbus00 committed:
changes made by angel

catbus00 committed:
Fix date issue

catbus00 committed:
Use correct type for activity

catbus00 committed:
Env changes

catbus00 committed:
Apply changes from Mo's branch

catbus00 committed:
Merge branch 'dev' into 46_Create_Events_Page

# Conflicts:
#	package-lock.json
#	src/App.tsx
#	src/components/AccountCreation/AccountCreation.tsx
#	src/components/AccountCreation/AccountCreationContainer.tsx
#	src/components/AuthenticatedContent/AuthenticatedContent.tsx

catbus00 committed:
Fix imports

catbus00 committed:
Escape apostrophe

catbus00 committed:
Add success / fail message after form is submitted

catbus00 committed:
Meeting troubleshooting update

catbus00 committed:
Added MUI-friendly code

catbus00 committed:
Add new groupID that Angel gave me

angelkbrown committed:
Merge branch 'dev' into 46_Create_Events_Page

angelkbrown commented on src/components/AddEventToGroupForm/AddEventToGroupForm.tsx:
```diff
@@ -0,0 +1,179 @@
+        eventEndpoint,
+        {
+          name: eventName,
+          groupID: "657d0f422cb49ec9046b3b8c",
```
@catbus00, heads up that @marice-romero is working on implementing a link from the `/group/id` page to this form...we should be able to pass a group id that way so that we can populate that here. Very clever of you to hardcode what was missing just to get everything else working though! Great thinking! 💯

angelkbrown reviewed (approved):
Awesome work getting this working, @catbus00!! I know the group id is still hardcoded for the moment, but that's just a small detail and you've done all the hard work already! I really love that you found a way to work on this without having all the pieces lined up. Very scrappy and clever! Huge congrats on this! I know it was a lot! 🌟👏 🎉 

I tested this and it's working to create events for me! Admittedly, I've only given a cursory review now, but I think as long as things are working well enough, it's probably best to move quickly now and come back for any refactoring or tweaks later. 🚢

marice-romero reviewed (approved):
Merged this into the GroupView branch and passed the groupID to the CreateEvents page and it worked! Great job, @catbus00 - this was a huge task and it looks great!

dargaCode committed:
update package.lock

not sure why these changes are happening, might be some leftover detail from the merge conflict

dargaCode reviewed (approved):
Thank you Jillian! Great work with the types in this PR 📈
